src/main.rs

```diff
@@ -9,47 +9,111 @@ extern crate serde_dynamodb;
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct WorkflowTask {
+    id: String,
+    workflow_id: String,
+    status: String,
+    regulator: Regulator,
+}
+
 #[derive(Serialize)]
 struct RegulateResponse {
     id: String
 }
 
-#[post("/regulate")]
-fn regulate() -> Json<RegulateResponse> {
+#[derive(Deserialize)]
+struct RegulateData {
+    regulators: Vec<Regulator>,
+}
+
+#[post("/regulate", data = "<data>")]
+fn regulate(data: Json<RegulateData>) -> Json<RegulateResponse> {
+    let regulators = data.into_inner().regulators;
+
     let workflow_id = Uuid::new_v4();
     let workflow = Workflow {
         id: workflow_id.to_string(),
         status: "InProgress".to_owned(),
     };
-	let client = DynamoDbClient::new(Region::UsEast1);
 
+    // This block could be improved.
     match serde_dynamodb::to_hashmap(&workflow) {
         Ok(workflow_ddb) => {
             let mut put_item_input: PutItemInput = Default::default();
             put_item_input.item = workflow_ddb;
             put_item_input.table_name = "workflows".to_owned();
-            match client.put_item(put_item_input).sync() {
-                Ok(_output) => {
+            DDB.with(|client| {
+                match client.borrow_mut().put_item(put_item_input).sync() {
+                    Ok(_output) => {
+                    }
+                    Err(error) => {
+                        println!("Error: {:?}", error);
+                    }
                 }
-                Err(error) => {
-                    println!("Error: {:?}", error);
-                }
-            }
+            });
         },
         Err(err) => {
             println!("{:?}", err);
         }
     }
 
+    let workflow_tasks: Vec<WorkflowTask> = regulators.iter().map(|regulator| {
+        WorkflowTask {
+            id: Uuid::new_v4().to_string(),
+            workflow_id: workflow_id.to_string(),
+            status: "InProgress".to_owned(),
+            // Assuredly, this can be done more elegantly.
+            regulator: Regulator {
+                name: regulator.name.clone(),
+                context: regulator.context.clone(),
+            },
+        }
+    }).collect();
+
+    // Batch into a single DDB call.
+    for task in &workflow_tasks {
+        match serde_dynamodb::to_hashmap(&task) {
+            Ok(task_ddb) => {
+                let mut put_item_input: PutItemInput = Default::default();
+                put_item_input.item = task_ddb;
+                put_item_input.table_name = "tasks".to_owned();
+                DDB.with(|client| {
+                    match client.borrow_mut().put_item(put_item_input).sync() {
+                        Ok(_output) => {
+                        }
+                        Err(error) => {
+                            println!("Error: {:?}", error);
+                        }
+                    }
+                });
+            },
+            Err(err) => {
+                println!("{:?}", err);
```

Comment:
check out `log` and `env_logger` crates for simple and awesome logging

Reply:
Yeah, I'll set up logging at some point. I also need to improve the nested error handling like you mentioned. But hey, it does a thing!